src/main/java/ch/uzh/ifi/hase/soprafs22/service/UserService.java

```diff
@@ -157,4 +157,12 @@ public boolean isAvailable(User userInput) {
     User user = userRepository.findByUsername(userInput.getUsername());
     return user == null;
   }
+
+  public void deleteUser(long userId){
+      User userToDelete = this.userRepository.findById(userId);
+      if(userToDelete == null){
+          throw new ResponseStatusException(HttpStatus.NOT_FOUND, "user does not exit"); //Throws 404 if user with userId doesn't exist
+      }
+      userRepository.deleteById(userId);
```

Comment:
What happens if you just delete by id where the user doesn't exist (without getting the user and checking if it's null)? 
Do you have to flush the repo or is this done in deleteById?

Reply:
Not sure, will try to find out

Reply:
should be done